spec/unit/facter/util/infiniband_spec.rb

```diff
@@ -1,5 +1,4 @@
 require 'spec_helper'
-require 'facter/util/file_read'
```

Comment:
Why did this produce errors or do you have the errors handy?  I ran unit tests with Github Actions without this change and got no errors.

Reply:
I reproduced the error by fetching your latest upstream changes in master and running the rpec-tests again against the lastest master. I'm using PDK 2.2.0 and ran it with your Gemfile.lock. The error remains the same:

```
$ pdk test unit
pdk (WARN): This module is compatible with an older version of PDK. Run `pdk update` to update it to your version of PDK.
pdk (INFO): Using Ruby 2.7.3
pdk (INFO): Using Puppet 7.9.0
[✔] Installing missing Gemfile dependencies.
[✔] Preparing to run the unit tests.
The `rake` executable in the `rake` gem is being loaded, but it's also present in other gems (puppet-lint).
If you meant to run the executable for another gem, make sure you use a project specific binstub (`bundle binstub <gem_name>`).
If you plan to use multiple conflicting executables, generate binstubs for them and disambiguate their names.
/opt/puppetlabs/pdk/private/ruby/2.7.3/bin/ruby -I/opt/puppetlabs/pdk/share/cache/ruby/2.7.0/gems/rspec-core-3.10.1/lib:/opt/puppetlabs/pdk/share/cache/ruby/2.7.0/gems/rspec-support-3.10.2/lib /opt/puppetlabs/pdk/share/cache/ruby/2.7.0/gems/rspec-core-3.10.1/exe/rspec --pattern spec/\{aliases,classes,defines,functions,hosts,integration,plans,tasks,type_aliases,types,unit\}/\*\*/\*_spec.rb --format progress
No facts were found in the FacterDB for Facter v4.2.2 on {:operatingsystem=>"RedHat", :operatingsystemrelease=>"/^7/", :hardwaremodel=>"x86_64"}, using v3.14.1 instead
No facts were found in the FacterDB for Facter v4.2.2 on {:operatingsystem=>"RedHat", :operatingsystemrelease=>"/^8/", :hardwaremodel=>"x86_64"}, using v3.14.1 instead

An error occurred while loading ./spec/unit/facter/util/infiniband_spec.rb. - Did you mean?
                    rspec ./spec/unit/facter/infiniband_hcas_spec.rb
                    rspec ./spec/unit/facter/has_infiniband_spec.rb
                    rspec ./spec/unit/facter/infiniband_rate_spec.rb

Failure/Error: require 'facter/util/file_read'

LoadError:
  cannot load such file -- facter/util/file_read
# /opt/puppetlabs/pdk/share/cache/ruby/2.7.0/gems/rspec-puppet-2.10.0/lib/rspec-puppet/monkey_patches.rb:412:in `require'
# /opt/puppetlabs/pdk/share/cache/ruby/ 2.7.0/gems/rspec-puppet-2.10.0/lib/rspec-puppet/monkey_patches.rb:412:in `require'
# ./spec/unit/facter/util/infiniband_spec.rb:2:in `<top (required)>'
Run options: exclude {:bolt=>true}


Finished in 0.00007 seconds (files took 11.47 seconds to load)
0 examples, 0 failures, 1 error occurred outside of examples

/opt/puppetlabs/pdk/private/ruby/2.7.3/bin/ruby -I/opt/puppetlabs/pdk/share/cache/ruby/2.7.0/gems/rspec-core-3.10.1/lib:/opt/puppetlabs/pdk/share/cache/ruby/2.7.0/gems/rspec-support-3.10.2/lib /opt/puppetlabs/pdk/share/cache/ruby/2.7.0/gems/rspec-core-3.10.1/exe/rspec --pattern spec/\{aliases,classes,defines,functions,hosts,integration,plans,tasks,type_aliases,types,unit\}/\*\*/\*_spec.rb --format progress failed
```

Reply:
I think this is an issue with PDK, the unit tests in Github Actions don't use PDK.  That require line still works interactively with IRB:

```
2.5.5 :001 > require 'facter/util/file_read'
 => true 
```

This is with Facter 4 loaded in my environment.  If I had to guess, PDK does something it shouldn't to break loaded gems or something.

Reply:
Either way , that was to a spec file and I think a while ago removed the need for that and just forgot to remove the require line.